### PR TITLE
Diagnostics: Track google query methods

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This PR adds 3 new diagnostics events:
- GOOGLE_QUERY_SKU_DETAILS_REQUEST
- GOOGLE_QUERY_PURCHASES_REQUEST
- GOOGLE_QUERY_PURCHASE_HISTORY_REQUEST

These events will be tracked when hitting the corresponding Google query methods.

### CODE REVIEW

File: feature/google/src/bc5SupportDev/java/com/revenuecat/purchases/google/BillingWrapper.kt [hunk 1]
```diff
@@ -60,7 +61,9 @@ private const val RECONNECT_TIMER_MAX_TIME_MILLISECONDS = 1000L * 60L * 15L // 1
 class BillingWrapper(
     private val clientFactory: ClientFactory,
     private val mainHandler: Handler,
-    private val deviceCache: DeviceCache
+    private val deviceCache: DeviceCache,
+    @Suppress("unused")
+    private val diagnosticsTracker: DiagnosticsTracker?
```
Comment (tonidero) [inline feature/google/src/bc5SupportDev/java/com/revenuecat/purchases/google/BillingWrapper.kt:66]: I foresee a complicated merge conflict with BC5 once we support it... So I decided to just add the functionality to the BC4 version for now. My plan is to create a ticket to just do this again in the BC5 integration branch after it's been merged to main and to that branch. Lmk if you have any concerns over that.

File: feature/google/src/test/java/BillingWrapperTestBase.kt [hunk 2]
```diff
@@ -57,7 +64,14 @@ open class BillingWrapperTestBase {
 
     @Before
     fun setup() {
+        clearAllMocks()
+        storeProducts = null
+        purchasesUpdatedListener = null
+        billingClientStateListener = null
```
Comment (tonidero) [inline feature/google/src/test/java/BillingWrapperTestBase.kt:70]: We were not cleaning mocks for all these tests afaik, which is surprising... I'm not sure if mockk is clearing mocks automatically on its own on each test... But in any case, I thought clearing mocks manually was safer. I would actually prefer to recreate all the mocks on each test, but this should also work.

File: feature/google/src/unityIAP/java/com/revenuecat/purchases/google/BillingWrapper.kt [hunk 3]
```diff
@@ -704,7 +715,53 @@ class BillingWrapper(
...
+        diagnosticsTracker?.trackGoogleQueryPurchasesRequest(
+            skuType,
+            billingResult.responseCode,
+            billingResult.debugMessage,
+            responseTimeMillis = dateProvider.now.time - requestStartTime
+        )
+    }
+
+    private fun trackGoogleQueryPurchaseHistoryRequestIfNeeded(billingResult: BillingResult, requestStartTime: Long) {
+        diagnosticsTracker?.trackGoogleQueryPurchaseHistoryRequest(
```
Comment (tonidero) [inline feature/google/src/unityIAP/java/com/revenuecat/purchases/google/BillingWrapper.kt:761]: Currently we understand that diagnostics is disabled if this class is null. I think that's ok, but I'm not sure if it's explicit enough... Not planning to change it for now but lmk if you have any concerns.

File: purchases/src/main/kotlin/com/revenuecat/purchases/BillingFactory.kt [hunk 4]
```diff
@@ -7,23 +7,27 @@ import androidx.annotation.VisibleForTesting
...
 
+    @Suppress("LongParameterList")
     fun createBilling(
         store: Store,
         application: Application,
         backend: Backend,
         cache: DeviceCache,
-        observerMode: Boolean
+        observerMode: Boolean,
+        diagnosticsTracker: DiagnosticsTracker?
```
Comment (tonidero) [inline purchases/src/main/kotlin/com/revenuecat/purchases/BillingFactory.kt:24]: Not tracking amazon requests yet.

File: common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsTracker.kt [hunk 5]
```diff
@@ -37,6 +37,61 @@ class DiagnosticsTracker(
...
+        billingDebugMessage: String,
+        responseTimeMillis: Long
+    ) {
+        trackEvent(
+            DiagnosticsEntry.Event(
+                name = DiagnosticsEventName.GOOGLE_QUERY_PURCHASE_HISTORY_REQUEST,
+                properties = mapOf(
+                    "billing_response_code" to billingResponseCode,
+                    "billing_debug_message" to billingDebugMessage,
+                    "response_time_millis" to responseTimeMillis
```
Comment (NachoSoto) [inline common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsTracker.kt:89]: `"response_time_millis"` is repeated in all of these, but it seems like a common parameter.
Would it make sense to extract a constant?

**Review by NachoSoto — APPROVED** [review]

**Comment by tonidero** [comment]
Tests are failing due to Sdkman but they are passing locally and since this is being merged to a non-main branch, will merge it for now. Will deal with those issues in that branch.

**Pull request merged**